include/revng/Model/Helpers.h

```diff
@@ -16,32 +16,43 @@
 namespace model {
 
 template<typename Type>
-concept EntityWithKey = requires(Type &&Value) {
+concept EntityWithKey = requires(const Type &Value) {
   { Value.key() } -> std::convertible_to<const typename Type::Key &>;
 };
 
 template<typename Type>
-concept EntityWithCustomName = requires(Type &&Value) {
-  { Value.CustomName() } -> std::convertible_to<model::Identifier &>;
+concept EntityWithCustomName = requires(const Type &Value) {
+  { Value.CustomName() } -> std::convertible_to<const model::Identifier &>;
 };
 
 template<typename Type>
-concept EntityWithOriginalName = requires(Type &&Value) {
-  { Value.OriginalName() } -> std::convertible_to<std::string &>;
+concept EntityWithOriginalName = requires(const Type &Value) {
+  { Value.OriginalName() } -> std::convertible_to<const std::string &>;
 };
 
 template<typename Type>
-concept EntityWithComment = requires(Type &&Value) {
-  { Value.Comment() } -> std::convertible_to<std::string &>;
+concept EntityWithComment = requires(const Type &Value) {
+  { Value.Comment() } -> std::convertible_to<const std::string &>;
 };
 
 template<typename Type>
-concept EntityWithReturnValueComment = requires(Type &&Value) {
-  { Value.ReturnValueComment() } -> std::convertible_to<std::string &>;
+concept EntityWithReturnValueComment = requires(const Type &Value) {
+  { Value.ReturnValueComment() } -> std::convertible_to<const std::string &>;
 };
 
+template<typename Type>
+void ensureCompatibility() {
+  static_assert(EntityWithCustomName<Type> || EntityWithOriginalName<Type>
+                  || EntityWithComment<Type>
+                  || EntityWithReturnValueComment<Type>,
+                "This would be a no-op.");
+}
+
 template<typename LHS, typename RHS>
 LHS &copyMetadata(LHS &To, const RHS &From) {
```

Comment:
Can you add a comment (maybe at the top of the file?) defining what we mean by "metadata"?

Also, maybe this file should be renamed to something related to "metadata".

Reply:
> Can you add a comment (maybe at the top of the file?) defining what we mean by "metadata"?

👌 

> Also, maybe this file should be renamed to something related to "metadata".

It used to contain other helpers at some point, but they all got obsolete / moved someplace else.

Reply:
Done!